pipettes/firmware/motor_control.cpp

```diff
@@ -0,0 +1,95 @@
+#include "pipettes/firmware/motor_control.hpp"
+
+#include "FreeRTOS.h"
+#include "common/core/bit_utils.hpp"
+#include "firmware/common/motor.h"
+#include "stm32g4xx_hal.h"
+#include "task.h"
+
+using namespace motor_control;
+
+/*
+ * Motor Control and namespace.
+ * Private:
+ * build_command takes in a command address, a reference
+ * to the data to be transmitted and the output array to
+ * store this in.
+ * reset_data takes nothing and resets the global data variable
+ *
+ * Public Globals:
+ * data, the data returned from the driver
+ * status, the status of the driver
+ *
+ * Public:
+ * setup, sets the initial configurations of the driver
+ * set_speed, sets a global variable called speed
+ * get_speed, gets a global variable called speed
+ * move, moves the motor for an unlimited amount of time
+ * get_status, get the status of the GSTAT register
+ */
+
+/*
+ * Private Functions
+ */
+void MotorControl::build_command(uint8_t command, const uint32_t& command_data,
+                                 BufferType& txBuffer) {
+    // need to pass in data parameter and use int_to_bytes here
+    auto* output = txBuffer.begin();
```

Comment:
not sure if this is from lint, but `auto*` is appropriate when you're declaring something that _is_ a pointer, not just has pointer semantics. Sometimes it will be - if `BufferType` is an array - but sometimes it won't, sometimes it will just be an iterator, and then `auto*` will be incorrect. We should make it just `auto`.

Reply:
Really should just make this auto